### DISCUSSION TITLE
Too few arguments to function

### DISCUSSION THREAD

**Discussion by iislamtaleb** [livewire/livewire#8137] [4. General]
Hello, is this a bug or what because it's too weird!

1. First create an event the blade component 
`reserveButton.addEventListener('click',()=>{
		$wire.dispatch('make-reservation',{time:time});});`

2. Second listen for the event in the livewire class: 
` #[On('make-reservation')] 
    public function saveReservation($time)
    {   
        return redirect('/test')
        ->with('time', $time);
     }`
3. You will face "Too few arguments to function App\Livewire\ReserveAppointment::saveReservation(), 0 passed in" error.

**Comment by danie-ramdhani** [comment]
Here, try this example: https://wirebox.app/b/x6dmg

**Reply by iislamtaleb** [reply]
It's working fine, but when I add `return redirect('/test')` the error show up.

**Reply by danie-ramdhani** [reply]
try:

```php
session()->flash('time', $time);
$this->redirect('/');
```
